servicetalk-http-netty/src/test/java/io/servicetalk/http/api/ServiceTalkContentCodingTest.java

```diff
@@ -0,0 +1,250 @@
+import io.servicetalk.http.netty.HttpServers;
+import io.servicetalk.transport.api.HostAndPort;
+import io.servicetalk.transport.api.ServerContext;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.concurrent.api.Single.succeeded;
+import static io.servicetalk.encoding.api.ContentCodings.identity;
+import static io.servicetalk.http.api.CharSequences.contentEquals;
+import static io.servicetalk.http.api.HeaderUtils.encodingFor;
+import static io.servicetalk.http.api.HttpHeaderNames.ACCEPT_ENCODING;
+import static io.servicetalk.http.api.HttpHeaderNames.CONTENT_ENCODING;
+import static io.servicetalk.http.api.HttpResponseStatus.UNSUPPORTED_MEDIA_TYPE;
+import static io.servicetalk.http.api.HttpSerializationProviders.textDeserializer;
+import static io.servicetalk.http.api.HttpSerializationProviders.textSerializer;
+import static io.servicetalk.transport.netty.internal.AddressUtils.localAddress;
+import static io.servicetalk.transport.netty.internal.AddressUtils.serverHostAndPort;
+import static java.util.Arrays.stream;
+import static java.util.Collections.disjoint;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class ServiceTalkContentCodingTest extends BaseContentCodingTest {
+
+    private static final Function<Scenario, StreamingHttpServiceFilterFactory> REQ_VERIFIER = (scenario)
+            -> new StreamingHttpServiceFilterFactory() {
+        @Override
+        public StreamingHttpServiceFilter create(final StreamingHttpService service) {
+            return new StreamingHttpServiceFilter(service) {
+                @Override
+
+                public Single<StreamingHttpResponse> handle(final HttpServiceContext ctx,
+                                                            final StreamingHttpRequest request,
+                                                            final StreamingHttpResponseFactory responseFactory) {
+                    final ContentCodec reqEncoding = scenario.requestEncoding;
+                    final List<ContentCodec> clientSupportedEncodings = scenario.clientSupported;
+
+                    try {
+
+                        String requestPayload = request.payloadBody(textDeserializer())
+                                .collect(StringBuilder::new, StringBuilder::append)
+                                .toFuture().get().toString();
+
+                        assertEquals(payloadAsString((byte) 'a'), requestPayload);
```

Comment:
Assertions work only in `main` test thread. If you use assertion in offloaded thread or on a server side, they won't fail the test.
We usually use a `BlockingQueue` to save the result we observed on the server side and validate the values in the queue in the main test thread.

Reply:
Discussed offline, the errors are surfaced with the assertion of expected successful response + printed stack trace.
However, changing the logic to capture the error and re-throw under main, to help with readability.